spring-cloud-gateway-server-webflux/src/main/java/org/springframework/cloud/gateway/config/GatewayAutoConfiguration.java

```diff
@@ -924,6 +952,35 @@ public TokenRelayGatewayFilterFactory tokenRelayGatewayFilterFactory(
 
 	}
 
+	public static class GatewayServerWebfluxBeanPostProcessor implements BeanPostProcessor {
+
+		private final VersionProperties versionProperties;
+
+		public GatewayServerWebfluxBeanPostProcessor(VersionProperties versionProperties) {
+			this.versionProperties = versionProperties;
+		}
+
+		@Override
+		public @Nullable Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+			if (bean instanceof ApiVersionStrategy && beanName.equals("mvcApiVersionStrategy")) {
+				List<ApiVersionResolver> versionResolvers = new ArrayList<>();
+				String headerName = versionProperties.getHeaderName();
+				if (StringUtils.hasText(headerName)) {
+					versionResolvers.add(exchange -> exchange.getRequest().getHeaders().getFirst(headerName));
+				}
+
+				GatewayApiVersionStrategy strategy = new GatewayApiVersionStrategy(versionResolvers,
+						new SemanticApiVersionParser(), versionProperties.isRequired(), null, true, null);
```

Comment:
Could `ApiVersionParser` be a bean so it is configurable?

Reply:
Should the deprecation handler also be an optional bean?

Reply:
yup